x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/plan/logical/EsRelation.java

```diff
@@ -96,12 +96,14 @@ public List<Attribute> output() {
 
     @Override
     public boolean expressionsResolved() {
+        // For unresolved expressions to exist in EsRelation is fine, as long as they are not used in later operations
+        // This allows for them to be converted to null@unsupported fields in final output, an important feature of ES|QL
         return true;
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(index, esSourceOptions, frozen);
+        return Objects.hash(index, attrs, esSourceOptions, frozen);
```

Comment:
`EsRelation` was already ported to `esql` by Nhat - I think this fix should be applied to `esql`'s copy of that, not here.

Reply:
Curiously, in the QL version, it used attrs in the hashCode, but not in the equals method. My change adds it to equals. Nhat's version removed it from hashCode. I'm curious if his change fixed something else, that I will now break.

Reply:
Turns out the changes I made were no longer necessary, and only needed for an earlier version of union-types. I kept just one clarifying comment.